### PR TITLE
Rewrite GroupNavigation with tests 

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-13528
affects: Student
release note: Fixed some minor styling issues in group navigation

### CODE REVIEW

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>



#### Release Note: 
Fixed some minor styling issues in group navigation

#### Affected Apps: Student

[MBL-13528](https://instructure.atlassian.net/browse/MBL-13528)

<details><summary>:x: XCTest failed: StudentUnitTests/RoutesTests/testGroup</summary>


    Assertion Failure: RoutesTests.swift:87: XCTAssertNil failed: "<Student.GroupNavigationViewController: 0x7fbbd98cd070>"
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: Core/Core/Store/Store.swift [hunk 1]
```diff
@@ -127,7 +127,7 @@ public class Store<U: UseCase>: NSObject, NSFetchedResultsControllerDelegate {
         request(useCase, force: force, callback: callback)
     }
 
-    public func exhaust(while condition: @escaping (U.Response) -> Bool) {
+    public func exhaust(while condition: @escaping (U.Response) -> Bool = { _ in true }) {
```
Comment (bluwave) [inline Core/Core/Store/Store.swift]: nice

**Comment by bluwave** [comment]
A couple of times there was not title

IMG_3652
IMG_3653

**Comment by vanwagonet** [comment]
Thanks, I'll take a look & see if I can find why the name sometimes isn't showing up.

**Comment by bluwave** [comment]
yeah that is working nicely.

**Pull request merged**